contracts/minter-suite/Minters/MinterHolder/MinterHolderV2.sol

```diff
@@ -0,0 +1,765 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "../../../interfaces/0.8.x/IGenArt721CoreContractV3.sol";
+import "../../../interfaces/0.8.x/IMinterFilterV0.sol";
+import "../../../interfaces/0.8.x/IFilteredMinterHolderV0.sol";
+import "../../../interfaces/0.8.x/IDelegationRegistry.sol";
+
+import "@openzeppelin-4.5/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin-4.5/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin-4.5/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin-4.5/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin-4.5/contracts/utils/structs/EnumerableMap.sol";
+
+pragma solidity 0.8.17;
+
+/**
+ * @title Filtered Minter contract that allows tokens to be minted with ETH
+ * when purchaser owns an allowlisted ERC-721 NFT.
```

Comment:
Realizing that we should update the documentation here to more explicitly highlight that this minter does not "keep track" of if a given minter has already minted or not, it just allows one to use another ERC721 as a gate for minting without attempting to enforce a "snapshot" of ERC721 ownership (which is likely better served by an allowlist minter in most use cases).

Reply:
The 'ole Merkle minter is a pretty good one 😄 agree that over-explaining here is a good idea

Reply:
Sounds good! I added these two lines...what do ya'll think?

> * @title Filtered Minter contract that allows tokens to be minted with ETH
 * when purchaser owns an allowlisted ERC-721 NFT. This contract does NOT track
 * if a purchaser has/has not minted already -- it simply restricts purchasing
 * to anybody that holds one or more of a specified list of ERC-721 NFTs.

Reply:
That is perfect @mchrupcala 🙏